src/webserver.py

```diff
@@ -141,9 +149,11 @@ def update_config(self, flask_environment: dict[str, str], is_webserver_running:
                 "Webserver configuration check failed, "
                 "please review your charm configuration or database relation"
             ) from exc
+        self._database_migration.run(flask_environment)
```

Comment:
Yeh this doesn't seem to be the right place to do it.

I'm also wondering whether it is a flask or webserver responsibility for running the migration? It seems more a charm responsibility to execute the migration script and then start the webserver/ flask since they are essentially distinct responsibilities

Reply:
Discussed with @gregory-schiano, because the script gets the flask environment it seems to be appropriate to run it from the flask app class